iotfunctions/pipeline.py

```diff
@@ -2685,8 +2685,13 @@ def check_data_items_type(self, df, items):
                         # 'object'.
                         # Conversion via combination of where() and mask() is 60% faster than an approach with apply()
                         try:
-                            df_column_tmp = df_column.where(df_column.isna(), np.bool_(df_column))
-                            df[data_item['name']] = df_column_tmp.mask(df_column_tmp.isna(), None)
+                            if isinstance(df_column.values[0], str):
```

Comment:
Hi Shraddha,  I stumbled on operator precedence. Additional parenthesis make it easier to understand:
                                 cond = (df_column == 'True')   
You only check on string for the first entry in df_column. In general, the column could contain a mixture of everything, for example 'true', 'false', (python) None . In such a scenario, the result of the conversion is kind of sensitive to the order of entries because (python) None is not a string type.

Reply:
Thanks for the review Peter. With the changes I've made we go to the first conditional when we have one or more of 'True' 'False' 'None' or python None